src/api/java/com/minecolonies/api/entity/ai/statemachine/tickratestatemachine/TickRateStateMachine.java

```diff
@@ -40,26 +52,59 @@ public TickRateStateMachine(@NotNull final IAIState initialState, @NotNull final
     @Override
     public void tick()
     {
+        // Update the tickrate counter, skip tick if we're lower
+        tickRateCounter++;
+        if (tickRateCounter < tickRate)
+        {
+            return;
+        }
+        tickRateCounter = 0;
+
+        // Update the tick counter for transitions
         tickCounter++;
         if (tickCounter > MAX_TICKRATE)
         {
             tickCounter = 1;
         }
 
+        for (final ITickingTransition<S> transition:eventTransitionMap.get(AIBlockingEventType.AI_BLOCKING))
```

Comment:
For @OrionDevelopment we do this with for loops because the lists are very small and the streaming overhead is considerably (@someaddons did some tests and it  improved the tps considerably)